sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/directconnectivity/ReplicatedResourceClient.java

```diff
@@ -85,6 +94,19 @@ public ReplicatedResourceClient(
             serviceConfigReader,
             useMultipleWriteLocations);
         this.enableReadRequestsFallback = enableReadRequestsFallback;
+
+        switch (Configs.getSpeculationType()) {
+            case SpeculativeProcessor.THRESHOLD_BASED:
+                this.speculativeProcessor = new ThresholdBasedSpeculation();
+                break;
+            case SpeculativeProcessor.THOMPSON_SAMPLING_BASED:
+                this.speculativeProcessor = new ThompsonSamplingBasedSpeculation(this.transportClient
+                    .getGlobalEndpointManager()
+                    .getAvailableReadEndpoints());
```

Comment:
use readEndpoints instead?

I think the difference between availableEndpoints vs readEndpoints is:
availableEndpoints -> the regions configured for the account
readEndpoint -> the regions based on the PreferredRegions and some other criteria.

For speculative processing, we probably should still honor the preferredRegion etc config

Reply:
Thanks for the suggestion. Will use readEndpoints